topics/reference_guide/text_selection.md

```diff
@@ -20,36 +20,37 @@ See also:
 The two actions [Extend Selection and Shrink Selection](https://www.jetbrains.com/help/idea/working-with-source-code.html#editor_code_selection) in IntelliJ Platform IDEs let you adjust selected text based on the structure of the source code.
 This makes it easy to select not only expressions, blocks, and function definitions, but also code like whole lines or tags in JavaDoc comments.
 
-If you’re implementing a custom language, the IntelliJ Platform will provide you with basic implementations of this EP, allowing you to select code based on your PSI structure and to select whole lines.
+When implementing a custom language, the IntelliJ Platform provides a basic implementations of this EP, allowing you to select code based on your PSI structure and to select whole lines.
```

Comment:
implementation<s>

Reply:
I fixed that already to "provides basic implementations" since you get several for free AFAIK. Like `LineCommentSelectioner`, or `NaturalLanguageTextSelectioner` should work for most languages..